app/controllers/user_sessions_controller.rb

```diff
@@ -22,6 +22,8 @@ def create
       if @user_session.save
         flash[:notice] = ts("Successfully logged in.")
         @current_user = @user_session.record
+        REDIS_GENERAL.sadd("last_login_list", params[:user_session][:login])
+        REDIS_GENERAL.set("last_login_#{params[:user_session][:login]}", Time.now.utc)
```

Comment:
Sorry I have had another thought reading this first thing in the morning with freshly caffeinated brain: shouldn't we use user id instead? Then if a user changes their name we still have their correct last login time.